docs/blog/2024-12-04-angular-state-management-2024.md

```diff
@@ -0,0 +1,143 @@
+
+Finally, the way we interact with our state is different. Instead of manipulating the state directly, we use the `patchState` function instead. `patchState` takes the state we want to manipulate, and uses a function to return a new version of that state. To add a new item to our `items` object, we can simply use the spread operator. Removing an item means we use `filter`, and updating an item means we use `map`. What's great about this is not only are we doing things in an immutable way, we're also getting all the types from our state. If we pass along a type that our state doesn't recognize, we'll get a type error before we even save.
+
+## Signal Store
+
+So Signal State is a more prescriptive way of handling smaller state, be it in a component or service. What is Signal Store all about? Signal Store is the more robust solution that you would expect for NgRx. It still is based on Signals, keeps the structure that most larger teams would want for this state solutions. Again, let's rework our previous example and update it for Signal Store.
+
+```ts
+const AppStore = signalStore(
+  withState<Store>({
+    items: [],
+  }),
+  withMethods((state) => ({
+    addToStore(item: StoreItem) {
+      patchState(state, (oldState) => ({
+        ...oldState,
+        items: [...oldState.items, item],
+      }));
+    },
+    removeFromStore(item: StoreItem) {
+      patchState(state, (oldState) => ({
+        ...oldState,
+        items: oldState.items.filter((e) => e.id !== item.id),
+      }));
+    },
+    updateStore(item: StoreItem) {
+      patchState(state, (oldState) => ({
+        ...oldState,
+        items: oldState.items.map((e) =>
+          e.id === item.id ? { ...item, name: 'bar' } : e
+        ),
+      }));
+    },
+  }))
+);
+```
+
+Here, we're staring to see a more structure approach to managing state that isolates our state interactions from the rest of our app. Instead of creating a service, we use the `signalStore` function instead. `signalStore` will return an injectable service instead that we provide to a component, or our root app instance. From here, we pass a `withState` function to provide any actual state value to the store. Like `signalState`, this is an object/record only.
+
+For modifying our store, we can use the `withMethods` function and pass any methods we want to expose to our app. What stands out here is that our stores value is accessible without having to inject it. Similar to `signalState`, we use the `patchState` to make any changes we need. Since the mechanism to modify the store in `signalStore` is very close to what we had in `signalState`, it's very approachable to migrating from your simple local store, to something more full featured. So if you're app and team grow significantly, there is a great path forward.
```

Comment:
```suggestion
For modifying our store, we can use the `withMethods` function and pass any methods we want to expose to our app. What stands out here is that our store's value is accessible without having to inject it. Similar to `signalState`, we use the `patchState` to make any changes we need. Since the mechanism to modify the store in `signalStore` is very close to what we had in `signalState`, it's very approachable when migrating from your simple local store to something more full featured. So if your app and team grow significantly, this is a great path forward.
```